jmbitcoin/jmbitcoin/secp256k1_transaction.py

```diff
@@ -146,7 +146,7 @@ def serialize(tx):
         o.write(b'\x01')
     o.write(num_to_var_int(len(txobj["ins"])))
     for inp in txobj["ins"]:
-        if len(inp["outpoint"]["hash"]) == 64:
+        if len(inp["outpoint"]["hash"]) == 64 or isinstance(inp["outpoint"]["hash"], unicode):
```

Comment:
why is this needed?

Reply:
seems length checks weren't reliable in some cases if the call sites sent unicode

Reply:
do you have an example input?

Reply:
Without this `test_mktx` in `jmclient/test/test_tx_creation.py` fails.

Reply:
for `'hash': 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'`

Reply:
that's a broken test then which should be fixed

Reply:
I've refactored this a bit, most of the type detection throughout the codebase is based checking for `bytes` now(although I've left some secondary regex and length based detection as a backup for now).

Reply:
I still don't think there can be anything valid that is not len() == 64. It'll create an invalid tx at best.